docs/devel/storage.md

```diff
@@ -1,3 +1,104 @@
+cluster/cli.sh ssh node01 -- sudo mkdir /var/run/kubevirt/hostpath
+
+## deploy hostPath PVC provisioner
+
+We are using the hostPath provisioner from [here](https://github.com/MaZderMind/hostpath-provisioner). Simply run from :
+
+```bash
+cluster/kubectl.sh apply -f docs/devel/hostpath-provisioner.yaml
+```
+
+This will create
+- hostpath-provisioner Service Account
+- hostpath-provisioner Cluster Role
+- hostpath-provisioner Cluster Role Binding
+- hostpath-provisioner Deployment
+- Create hostpath Storage Class, set as the default, so any Persistent Volume Claims (PVCs) will be serviced by this storage class
+
+After a few moments the hostpath-provisioner pod should be running, we are now ready to create PVCs against it. Any requested PVCs will be stored in /var/run/kubevirt/<pvc-name>-<hash>. The PV retention policy will be delete, which means that when you delete the PVC, the matching PV is also removed and the directory created on the host is removed as well.
+
+## deploy CDI controller
+
+The Containerized Data Importer controller is a controller that watches for PVCs with a specific annotation. If that annotation is detected, it will use the 'storage.image.endpoint' URL to download the image, convert it if needed and write it to the requested PVC. One can then use the PVC to start a VM using the image in it. To deploy the CDI controller run this:
+
+```bash
+cluster/kubectl.sh apply -f https://raw.githubusercontent.com/kubevirt/containerized-data-importer/master/manifests/controller/cdi-controller-deployment.yaml
+```
+
+This will create
+- cdi-sa Service Account
+- cdi Cluster Role
+- cdi-sa Cluster Role Binding
+- cdi-deploy Deployment
+
+After a few moments the cdi-deployment pod should be running, we are now ready to import images using a PVC annotation.
+
+## Create a PVC with annotation
+
+Now that we can request PVCs and we have the CDI controller running, we are ready to import an image. An example yaml file looks like this:
+```YAML
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: "image-pvc"
+  labels:
+    app: containerized-data-importer
+  annotations:
+    kubevirt.io/storage.import.endpoint: "https://download.cirros-cloud.net/0.4.0/cirros-0.4.0-x86_64-disk.img"   # Required.  Format: (http||s3)://www.myUrl.com/path/of/data
+    kubevirt.io/storage.import.secretName: "" # Optional.  The name of the secret containing credentials for the data source
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 20Mi
+```
+
+As you can see we are creating a PVC named 'image-pvc' with 2 annotations. In this case we are requesting the cirros image because its a small image, but any http or s3 end point will do. If you need to provide a secret the second annotation can be used to provide it. In our example we don't need a secret.
+
+We CDI controller will see the annotation and spawn an 'importer' pod that does the actual work. Once the pod is finished there will be a disk.img in the PVC.
```

Comment:
So is the download done before the PVC is bound, or after? If before, is this PVC then really meant for vm consumption or is it more like a golden image which can then be used to clone from into different PVCs? I am asking because it sounds to me like I have to wait for the PVC to be populated before I can start a vm.

With other words, is the image download part of the PVC provision step or afterwards, when the PVC is `bound`, and therefore maked as ready to be consumed by pods?

Reply:
@vatsalparekh FYI, regarding to our IRC conversation.

Reply:
The PVC is bound, with an annotation (the source image URL and secret), the CDI controller sees the annotation, which then spawns an importer pod to populate the bound PVC. The PVC has to be able to be consumed otherwise the importer pod can't write to it.

As soon as the PVC is bound it can be consumed by pods, however the contents may not be populated fully. The importer pod adds an annotation to the PVC indicating the status of the import (Failed/Ongoing/Success)